### PR TITLE
Fix incremental perf regression unsafety checking

### PR DESCRIPTION
**Pull request by oli-obk**
Perf regression introduced in #96294

We will simply avoid emitting the name of the unsafe function in MIR unsafeck, since we're moving to THIR unsafeck anyway.

### CODE REVIEW

File: compiler/rustc_mir_build/src/check_unsafety.rs [hunk 1]
```diff
@@ -546,75 +546,57 @@ enum UnsafeOpKind {
...
-            AssignToDroppingUnionField => "assignment to union field that might need dropping",
-            AccessToUnionField => "access to union field",
-            MutationOfLayoutConstrainedField => "mutation of layout constrained field",
-            BorrowOfLayoutConstrainedField => {
-                "borrow of layout constrained field with interior mutability"
-            }
-            CallToFunctionWith(..) => "call to function with `#[target_feature]`",
-        }
-    }
-
```
Comment (Emilgardis) [inline compiler/rustc_mir_build/src/check_unsafety.rs:566]: Why is this removed? This was used to only have a simple span label, and not a repeat of what is in the description

File: src/test/ui/async-await/async-unsafe-fn-call-in-safe.thir.stderr [hunk 2]
```diff
@@ -2,15 +2,15 @@ error[E0133]: call to unsafe function `S::f` is unsafe and requires unsafe funct
...
 
 error[E0133]: call to unsafe function `f` is unsafe and requires unsafe function or block
-  --> $DIR/async-unsafe-fn-call-in-safe.rs:15:5
+  --> $DIR/async-unsafe-fn-call-in-safe.rs:17:5
    |
 LL |     f();
-   |     ^^^ call to unsafe function
+   |     ^^^ call to unsafe function `f`
    |
    = note: consult the function's documentation for information on how to avoid undefined behavior
```
Comment (Emilgardis) [inline src/test/ui/async-await/async-unsafe-fn-call-in-safe.thir.stderr:15]: see here

File: compiler/rustc_mir_build/src/check_unsafety.rs (hunk 1 above)
Comment (oli-obk) [reply compiler/rustc_mir_build/src/check_unsafety.rs]: you're right, sorry, I got overzealous in reverting

**Review by compiler-errors — APPROVED** [review]
> We will simply avoid emitting the name of the unsafe function in MIR unsafeck, since we're moving to THIR unsafeck anyway.

I think this change is reasonable, since the highlighted span elucidates the call anyways.

**Pull request merged**